lib/modules/metabolic_syndrome.rb

```diff
@@ -221,13 +284,88 @@ def update_diabetes(severity,time,entity)
+        end
+      end
+
+      rule :monotherapy, [:diet_and_exercise], [:bitherapy] do |time,entity|
+        if entity[:diabetes] && entity[:diabetes][:severity]==1
+          entity[:medications] = Hash.new if entity[:medications].nil?
+          # delete medications above this stage...
+          [:glp1ra,:sglt2i,:basal_insulin,:prandial_insulin].each{|m| entity[:medications].delete(m)}
+          # prescribe metformin if it isn't already there...
+          entity[:medications][:metformin] = [ time, :diabetes ] if entity[:medications][:metformin].nil?
+        end
+      end
+
+      rule :bitherapy, [:monotherapy], [:tritherapy,:insulin] do |time,entity|
+        if entity[:diabetes] && entity[:diabetes][:severity]==2
+          entity[:medications] = Hash.new if entity[:medications].nil?
+          # delete medications above this stage...
+          [:sglt2i,:basal_insulin,:prandial_insulin].each{|m| entity[:medications].delete(m)}
+          # prescribe metformin and glp1ra if they aren't already there...
+          entity[:medications][:metformin] = [ time, :diabetes ] if entity[:medications][:metformin].nil?
+          entity[:medications][:glp1ra] = [ time, :diabetes ] if entity[:medications][:glp1ra].nil?
+        end
+      end
+
+      rule :tritherapy, [:bitherapy], [:insulin] do |time,entity|
+        if entity[:diabetes] && entity[:diabetes][:severity]==3
+          entity[:medications] = Hash.new if entity[:medications].nil?
+          # delete medications above this stage...
+          [:basal_insulin,:prandial_insulin].each{|m| entity[:medications].delete(m)}
+          # prescribe metformin and cocktail if they aren't already there...
+          entity[:medications][:metformin] = [ time, :diabetes ] if entity[:medications][:metformin].nil?
+          entity[:medications][:glp1ra] = [ time, :diabetes ] if entity[:medications][:glp1ra].nil?     
+          entity[:medications][:sglt2i] = [ time, :diabetes ] if entity[:medications][:sglt2i].nil?  
+        end
+      end
+
+      rule :insulin, [:tritherapy], [:insulin] do |time,entity|
+        if entity[:diabetes] && entity[:diabetes][:severity]==4
+          entity[:medications] = Hash.new if entity[:medications].nil?
+          # prescribe metformin and cocktail if they aren't already there...
+          entity[:medications][:metformin] = [ time, :diabetes ] if entity[:medications][:metformin].nil?
+          entity[:medications][:glp1ra] = [ time, :diabetes ] if entity[:medications][:glp1ra].nil?     
+          entity[:medications][:sglt2i] = [ time, :diabetes ] if entity[:medications][:sglt2i].nil?  
+          # prescribe insulin
+          if entity[:medications][:basal_insulin]
+            entity[:medications].delete(:basal_insulin)
+            entity[:medications][:prandial_insulin] = [ time, :diabetes ]            
+          else
+            entity[:medications][:basal_insulin] = [ time, :diabetes ]
+          end     
```

Comment:
This previous `if`/`else`/`end` block gets triggered (for diabetes severity of 4) with every rule cycle (currently weekly).  Based on the current logic, this would cause the following to happen:

* week 1: add `:basil_insulin`
* week 2: remove `:basil_insulin` and add `:prandial_insulin`
* week 3: add `:basil_insulin` (again)
* week 4: remove `basil_insulin` and add `:prandial_insulin` (again)

_rinse, lather, repeat_